frontend/src/store/playerElement.ts

```diff
@@ -0,0 +1,179 @@
+import { cloneDeep } from 'lodash-es';
+import { nextTick, reactive } from 'vue';
+import { watchOnce } from '@vueuse/core';
+// @ts-expect-error - No types on libass-wasm
+import SubtitlesOctopus from '@jellyfin/libass-wasm';
+import subtitlesOctopusWorkerUrl from '@jellyfin/libass-wasm/dist/js/subtitles-octopus-worker.js?url';
+import { mediaElementRef, playbackManagerStore } from '@/store';
+import { useRouter } from '@/composables';
+
+const playbackManager = playbackManagerStore();
+let subtitlesOctopus: SubtitlesOctopus | undefined;
+
+/**
+ * == INTERFACES ==
+ */
+interface PlayerElementState {
+  isFullscreenMounted: boolean;
+  isPiPMounted: boolean;
+  isStretched: boolean;
+}
+
+/**
+ * == STATE VARIABLES ==
+ */
+const defaultState: PlayerElementState = {
+  isFullscreenMounted: false,
+  isPiPMounted: false,
+  isStretched: true
+};
+
+const state = reactive<PlayerElementState>(cloneDeep(defaultState));
+
+/**
+ * == CLASS CONSTRUCTOR ==
+ */
+class PlayerElementStore {
+  /**
+   * == GETTERS ==
+   */
+  public get isFullscreenMounted(): boolean {
+    return state.isFullscreenMounted;
+  }
+
+  public set isFullscreenMounted(newIsMounted: boolean) {
+    state.isFullscreenMounted = newIsMounted;
+  }
+
+  public get isPiPMounted(): boolean {
+    return state.isPiPMounted;
+  }
+
+  public set isPiPMounted(newIsPipMounted: boolean) {
+    state.isPiPMounted = newIsPipMounted;
+  }
+
+  public get isStretched(): boolean {
+    return state.isStretched;
+  }
+
+  public set isStretched(newisStretched: boolean) {
+    state.isStretched = newisStretched;
+  }
+
+  public get isMinimized(): boolean {
+    return useRouter().currentRoute.value.fullPath !== '/playback/video';
+  }
+
+  /**
+   * == ACTIONS ==
+   */
+  public toggleMinimize = (): void => {
+    /**
+     * Destroys SSO before chaning view cause the canvas needs to be destroyed to be recreated in the other view
+     */
+    playerElement.freeSsaTrack(true);
+
+    const router = useRouter();
+
+    if (this.isMinimized) {
+      router.push('/playback/video');
+    } else {
+      router.replace(
+        typeof router.options.history.state.back === 'string'
+          ? router.options.history.state.back
+          : '/'
+      );
+    }
```

Comment:
This is the 3rd time I see this logic. Again, should be possible to have all of this handled just in a watcher in the playback page.